crates/flox/src/commands/environment.rs

```diff
@@ -52,11 +52,51 @@ impl Edit {
 
         let mut environment =
             resolve_environment(&flox, self.environment.as_deref(), "edit").await?;
-        let mut temporary_environment = environment.make_temporary()?;
-
         let nix = flox.nix(Default::default());
 
-        if let Some(file) = self.file {
+        match self.provided_manifest_contents()? {
+            // If provided with the contents of a manifest file, either via a path to a file or via
+            // contents piped to stdin, use those contents to try building the environment.
+            Some(new_manifest) => {
+                environment.edit(&nix, &flox.system, new_manifest).await?;
+                Ok(())
+            },
+            // If not provided with new manifest contents, let the user edit the file directly
+            // via $EDITOR (as long as `flox edit` was invoked interactively).
+            None => {
+                let editor = std::env::var("EDITOR").context("$EDITOR not set")?;
```

Comment:
Might want to handle a few more cases like this:
https://github.com/flox/flox/blob/4da62348a2d499ee62ab3476f2a78fed7e5ca2e7/flox-bash/lib/utils.sh#L179-L180

Reply:
I think I would leave that as a future improvement given that this is the "simplistic" version. I can create a ticket for that so we don't lose track of it.

Reply:
Fine by me!

Reply:
https://github.com/flox/flox/issues/245